translate-engagement-data/site.yml

```
@@ -27,6 +27,20 @@
           file:
             path: "{{ directory }}/ocp-init"
             state: directory
+        - name: Get allowed AZs per region and randomize the choice
+          set_fact:
+             choosen_zone: "{{ aws-region-zones[hosting_environments[0].ocp_cloud_provider_region] | random }}:
+          when:
+            - hosting_environments is defined
+            - hosting_environments|length > 0
+            - hosting_environments[0].ocp_aws_machineset_zones is not defined
+        - name: Set default AZ list if not already defined
+          set_fact:
+              hosting_environments[0].ocp_aws_machineset_zones[0]: "{{ hosting_environments[0].ocp_cloud_provider_region  }}"+"{{ choosen_zone }}"
```

This fails for me
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoneType: None
fatal: [localhost]: FAILED! => {"changed": false, "msg": "The variable name 'hosting_environments[0].ocp_aws_machineset_zones[0]' is not valid. Variables must start with a letter or underscore character, and contain only letters, numbers and underscores."}
```
If keeping the `hosting_environments` object is a requirement then the below change should do the trick. If not, then I'd suggest just changing the fact name to something like `aws_az`.
```suggestion
          set_fact:
              hosting_environments:
                - ocp_aws_machineset_zones:
                    - "{{ hosting_environments[0].ocp_cloud_provider_region  }}"+"{{ choosen_zone }}"
```